### ISSUE TITLE
Multi-printer support

### ISSUE THREAD

**Issue by thijstriemstra** [bofh69/nfc2klipper#12]
Afaik only a single printer is supported right now, it's not to possible to send a spool change to a printer other than the one defined in the config.

**Comment by bofh69** [comment]
Yes, the idea was to have a nfc-reader at the printers spool holder to automatically handle changes as you load the new filament.

What is your idea and workflow here?

Having a single nfc-reader where you tap the spool before mounting it on one of the printers? In that case some more input is needed to the system. Perhaps a tag that represent the printer the spool is moved to? That could also be used when storing a spool to update the location in spoolman.

Support more nfc-readers from the program? Then it is perhaps better to add an environment variable for choosing the configuration file and then just run it once per nfc-reader?

**Comment by thijstriemstra** [comment]
I bought a roll of ntag215 stickers and made magnetic tags that attach to my spoolbox: ![image](https://github.com/user-attachments/assets/81aa6852-841a-45c5-ad70-04bdb971c514)

at first i thought of making a rpi-based handscanner but this would need a long cable or battery pack. Since my printers are on same table, a scanner that is mounted to the wall is another idea since i can easily take out the magnetic tags, scan it and place it back.

The OpenSpool project uses an esp32 to control the np532 device and use wifi to transmit the data. These esp32 devices are much cheaper and smaller, more suitable than an rpi IMO.

So what about making a wall-mounted scanner, using esp32 (and maybe even micropython instead of c++)?

I would prefer if we also use YAML instead of json config files, which all klipper related projects seem to use.

Klipperscreen, for example, allows you to define an x nr of printers in the config, something like this:

```
[printer ender]
moonraker_host: ender.lan

[printer prusa]
moonraker_host: prusa.lan
```

**Comment by bofh69** [comment]
Nice setup!

I don't think nfc2klipper would be the right place for such a feature/workflow. Its probably better to add klipper/spoolman support to the OpenSpool project instead. They've already got the hardware and a lot of the firmware needed for it.


My klipper doesn't use YAML. It is a custom format that looks like toml/Windows INI-files. Switching to toml instead of json5 should be pretty straight forward and its probably a good idea to be more similar to the wider klipper ecosystem.

**Comment by bofh69** [comment]
Have a look at this project, it might be what you are looking for:
https://github.com/ManuelW77/Filaman
